pkgs/development/tools/rust/cargo-espflash/default.nix

```diff
@@ -15,8 +15,10 @@ rustPlatform.buildRustPackage rec {
     pkg-config
   ];
 
-  buildInputs = [
+  buildInputs = lib.optionals stdenv.isLinux [
```

Comment:
Hm, it feels a little "unclean" to introduce a second stdenv in a rust build. It's only used for platform checks of course but still.

Perhaps you could instead derive it from `rustPlatform.rust.rustc.stdenv` but I'm not sure that's better.

Reply:
Yeah but if you override rustPlatform, that doesn't override stdenv. Not an issue for a mere platform check for darwin but it doesn't feel clean.